### DISCUSSION TITLE
Slug system is not working as example

### DISCUSSION THREAD

**Discussion by intelcoder** [strapi/strapi#6147] [Legacy Topics]
I like to generate meta data from title and description but before I just tried slug system that I found from documentation. However, the example does not do anything and also throw error on getUpdate is not defined.

I believe that example is for the mongoose but I am not using any db. Is there anyway I can do make things work or has anyone tried ?

[doc example link](https://strapi.io/documentation/3.0.0-beta.x/guides/slug.html#auto-create-update-the-slug-attribute)

```
`const slugify = require('slugify');

module.exports = {
  beforeSave: async model => {
    if (model.title) {
      model.slug = slugify(model.title);
    }
  },
  beforeUpdate: async model => {
    if (model.getUpdate() && model.getUpdate().title) {
      model.update({
        slug: slugify(model.getUpdate().title),
      });
    }
  },
};`
```

**Comment by robertpiosik** [comment]
I think this guide is deprecated, it was written before UID field was introduced.

Please take a look at a UID field during Content-Type creation and from linked tutorial, look at the last step and simply replace `slug` with `uid`.

**Reply by intelcoder** [reply]
I am doing this because uid does not seem to support utf-8 and I also need to do similar things for page meta data. Do you know any example I can take a look ?
